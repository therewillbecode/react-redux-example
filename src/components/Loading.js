import React, { Component } from "react";

class Loading extends Component {
  componentDidMount() {
    this.props.handleAuthentication();
  }

  render() {
    const style = {
      position: "absolute",
      display: "flex",
      justifyContent: "center",
      height: "100vh",
      width: "100vw",
      top: 0,
      bottom: 0,
      left: 0,
      right: 0,
      backgroundColor: "white"
    };

    return (
      <div style={style}>
        <h1>Authenticating ... </h1>
        <img src="img/loading.svg" alt="loading" />
      </div>
    );
  }
}

export default Loading;
