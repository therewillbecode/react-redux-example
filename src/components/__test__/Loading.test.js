/* eslint-disable */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";

import Loading from "../Loading";

describe("Loading", () => {
  it("renders correctly", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Loading />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
