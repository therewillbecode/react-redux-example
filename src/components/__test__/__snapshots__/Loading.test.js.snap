// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "bottom": 0,
      "display": "flex",
      "height": "100vh",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "width": "100vw",
    }
  }
>
  <img
    alt="loading"
    src="img/loading.svg"
  />
</div>
`;
